app/integration/api/src/main/java/io/syndesis/integration/api/IntegrationResourceManager.java

```diff
@@ -216,7 +216,19 @@ default String sanitize(String name) {
     }
 
     default Collection<Dependency> collectDependencies(Integration integration) {
-        return collectDependencies(integration.getFlows().stream().flatMap(flow -> flow.getSteps().stream()).collect(Collectors.toList()), true);
+        Collection<Dependency> collectedDependencies = collectDependencies(integration.getFlows());
+        collectedDependencies.addAll(
+            collectDependencies(integration.getFlows().stream().flatMap(flow -> flow.getSteps().stream()).collect(Collectors.toList()), true)
+        );
+        return collectedDependencies;
+    }
+
+    default Collection<Dependency> collectDependencies(List<Flow> flows){
+        final List<Dependency> dependencies = new ArrayList<>();
+        for(Flow flow : flows){
+            dependencies.addAll(flow.getDependencies());
```

Comment:
Wonder if collecting dependencies of steps should be done here instead of in `collectDependencies(Integration)` to respect the hierarchy

Reply:
Good point

Reply:
I've amended with the suggestion, please have a look now